openedx/core/djangoapps/content/learning_sequences/api/tests/test_outlines.py

```diff
@@ -299,76 +312,81 @@ def get_sequence_keys(self, exclude=None):
         return [key for key in self.all_seq_keys if key not in exclude]
 
     def test_before_course_starts(self):
-        staff_details, student_details = self.get_details(
-            datetime(2020, 5, 9, tzinfo=timezone.utc)
-        )
-        # Staff can always access all sequences
-        assert len(staff_details.outline.accessible_sequences) == 5
-        # Student can access nothing
-        assert len(student_details.outline.accessible_sequences) == 0
-
-        # Everyone can see everything
-        assert len(staff_details.outline.sequences) == 5
-        assert len(student_details.outline.sequences) == 5
+        with override_waffle_flag(COURSE_ENABLE_UNENROLLED_ACCESS_FLAG, True):
```

Comment:
I do not know this for certain, but could `override_waffle_flag` be called in the (currently not included) `setUp` method so that it is run individually for every test instead of duplicating the code in each individual test? And if we are overriding this in every tests anyways, could this be put in `setUpTestData`?

Reply:
@pcockwell I couldn't get it to work by putting it in `setUpTestData` and I don't think it would work in `setUp` either. In `django-waffle`'s documentation they only show two methods i.e using context manager (`with` block) and decorator method.

BTW, I start using the decorator method as it look better i.e reduce 1 indentation level.

Reply:
I think the decorator would be more ideal. I'm just not sure whether or not the change to the waffle flag would be undone after the test is run if you do it in `setUp`. Now that I think of it, only database changes are impacted, so probably not...

Reply:
I just noticed the decorator usage now. Thanks